apps/developer-hub/src/components/Pages/DocumentationPage/index.tsx

```diff
@@ -0,0 +1,9 @@
+import { BasePage } from "../BasePage";
+
+export async function DocumentationPage(props: {
+  params: Promise<{ section: string; slug: string[] }>;
+}) {
+  const params = await props.params;
+  params.slug.unshift(params.section);
+  return BasePage({ params });
+}
```

Comment:
2 things:

1. We shouldn't be mutating params ever
2. We need to use jsx to invoke `BasePage` -- there are a ton of things both in nextjs and in react that will only work properly if using jsx (or `createElement` etc) to invoke components

So this should be written as:

```tsx
const params = await props.params;
return <BasePage params={{ ...params, slug: [ params.section, ...params.slug ] }} />
```

The same comment applies to the other pages too